### PR TITLE
[ENH] KILL WITH FIRE

### LINKED ISSUE #37: [ENH] Investigate/improve performance of data checks and pd.MultiIndex operations/iterations
**Is your feature request related to a problem? Please describe.**
There seems to be quite a problem related to performance of some data checks and `pd.MultiIndex` operations.

**Describe the solution you'd like**
Related: https://github.com/sktime/sktime/issues/4139

### PR DESCRIPTION
**Pull request by TonyBagnall**
#### Reference Issues/PRs

closes #37 #42 #99 #136 #240 #265 #291 #337 #341 #1388 #1491 #1629 #1446 #1156 #1734 #217 #2015

probably #599 too

#### What does this implement/fix? Explain your changes.

This is the big refactor/deprecation prior to 1.0.0, aka the project internally referred to as KILL WITH FIRE.

Removes forecasting module and datatypes module and all accompanying tests/references etc. Removes operator overloading, so called vectorisation and all output conversion from  BaseTransformer

*To Note*

1. BaseTransformer is linked into the interval trees, needs untangling to remove BaseTransformer completely, although we might want to keep it for tags

2. The registry will need some refactoring after, I'm just removing tests for now, could be the whole thing may be reworked. I think we should revisit tags

### CODE REVIEW

**Review by ltsaprounis — APPROVED** [review]
L
G
T
M

PS: Best PR of all time 
image

**Review by hadifawaz1999 — APPROVED** [review]
LGTM x10

**Review by MatthewMiddlehurst — DISMISSED** [review]
made some changes

https://github.com/aeon-toolkit/aeon/pull/2037/files/f9e4a98747a2d55b79f08ae75d5a5a2abcb4753d..0b8f227e5ebec3f4d1847539b1f2ea014441c675

File: pyproject.toml [hunk 1]
```diff
@@ -57,37 +57,37 @@ dependencies = [
...
     "stumpy>=1.5.1",
     "ruptures>=1.1.9",
-    "tbats>=1.1.0",
+#    "tbats>=1.1.0",
     "tensorflow>=2.14; python_version < '3.12'",
     "torch>=1.13.1",
     "tsfresh>=0.20.0",
     "tslearn>=0.5.2",
-    "xarray",
+#    "xarray",
```
Comment (chrisholder) [inline pyproject.toml:82]: Why are we just commenting out and not deleting entirely?

**Review by chrisholder — APPROVED** [review]
:fire:

**Review by hadifawaz1999 — APPROVED** [review]
PRESS IT

File: pyproject.toml (hunk 1 above)
Comment (TonyBagnall) [reply pyproject.toml:82]: good point, I was just testing it, but can remove after

**Review by chrisholder — APPROVED** [review]
:fire: :fire: :fire:

**Pull request merged**